nix/jormungandr.nix

```diff
@@ -0,0 +1,39 @@
+############################################################################
+# Packages for jormungandr and jcli.
+#
+# These use the functions from iohk-nix to build the version that we
+# require for cardano-wallet.
+#
+# To change the version:
+# 1. Adjust the "version" and/or "rev" variables below.
+# 2. Then change a single digit in *both* sha256 and cargoSha256.
```

Comment:
Actually when I changed the version and hashes I got:

```
error: invalid base-32 hash '4fw3cl2rxnw9mww1b1z96x2iapwbpdgyp4ra19dhvfzmlvaiml5j'
```
(so no new hash displayed)

when I did not update the hash (only updated version) I got:
```
fixed-output derivation produced path '/nix/store/6ib16q9kaq9wifr26cy9i1gwvawa0lcw-jormungandr-0.3.1415-source' with sha256 hash '1kg20570ri3bj99gicpp2z272igj9s7m2qw0z1ndk60bxcyghi2x' instead of the expected hash '1fw3cl2rxnw9mww1b1z96x2iapwbpdgyp4ra19dhvfzmlvaiml5j'
```

So I suppose step 2 it not needed (?)

Reply:
Oh I knew you would do that. :grin: The first base-32 character is actually 1 bit of hash and 4 bits of padding. So the only valid values are 0 and 1. I have updated the procedure to say just flip the first bit.

Step 2 is needed in case you have the sources already cached locally. The cache is indexed by hash, so if you did have the sources cached, then changed the version, it wouldn't download the new version of the sources.